streamer/src/nonblocking/quic.rs

```diff
@@ -153,8 +153,10 @@ async fn setup_connection(
                 }
                 connection.set_max_concurrent_uni_streams(
                     VarInt::from_u64(
-                        ((stake as f64 / total_stake as f64) * QUIC_TOTAL_STAKED_CONCURRENT_STREAMS)
-                            as u64,
+                        (stake as u128)
+                            .saturating_mul(QUIC_TOTAL_STAKED_CONCURRENT_STREAMS as u128)
+                            .checked_div(total_stake as u128)
+                            .unwrap_or(0) as u64,
```

Comment:
```suggestion
                        u64::try_from(
                            u128::from(stake)
                                .saturating_mul(u128::from(QUIC_TOTAL_STAKED_CONCURRENT_STREAMS))
                                .checked_div(u128::from(total_stake))
                                .unwrap_or(0)
                        ).unwrap_or(QUIC_MAX_UNSTAKED_CONCURRENT_STREAMS)
```

let's avoid `as` while we're here and it's easy.  if that constant can be declared as `u128` without affecting anything else, I'd suggest that as well